packages/rapier/src/lib/components/Attractor/Attractor.svelte

```diff
@@ -0,0 +1,69 @@
+<script lang="ts">
+  import type { RigidBody } from '@dimforge/rapier3d-compat'
+  import { Mesh, useFrame, Object3DInstance } from '@threlte/core'
+  import { SphereBufferGeometry, MeshBasicMaterial, Vector3, Object3D } from 'three'
+  import { useRapier } from '../../hooks/useRapier'
+  import type { AttractorProperties } from '../../types/components'
+
+  export let position: AttractorProperties['position'] = undefined
+  export let strength: NonNullable<AttractorProperties['strength']> = 1
+  export let range: NonNullable<AttractorProperties['range']> = 50
+  export let showHelper: AttractorProperties['showHelper'] = false
+  export let gravityType: NonNullable<AttractorProperties['gravityType']> = 'static'
+  export let gravitationalConstant: NonNullable<
+    AttractorProperties['gravitationalConstant']
+  > = 6.673e-11
+  const { world } = useRapier()
+  const gravitySource = new Vector3()
+  let obj = new Object3D()
+
+  const calcForceByType = {
+    static: (s: number, m2: number, r: number, d: number, G: number): number => s,
+    linear: (s: number, m2: number, r: number, d: number, G: number) => s * (d / r),
+    newtonian: (s: number, m2: number, r: number, d: number, G: number) =>
+      (G * s * m2) / Math.pow(d, 2)
+  }
+
+  function applyImpulseToBodiesInRange() {
+    const impulseVector = new Vector3()
+    obj.getWorldPosition(gravitySource)
+
+    world.forEachRigidBody((body: RigidBody) => {
+      const { x, y, z } = body.translation()
+      const bodyV3: Vector3 = new Vector3(x, y, z)
+      const distance: number = gravitySource.distanceTo(bodyV3)
+      if (distance < range) {
+        let force = calcForceByType[gravityType](
+          strength,
+          body.mass(),
+          range,
+          distance,
+          gravitationalConstant
+        )
+        // Prevent wild forces when Attractors collide
+        force = force === Infinity ? strength : force
+        impulseVector.subVectors(gravitySource, bodyV3).normalize().multiplyScalar(force)
+        body.applyImpulse(impulseVector, true)
+      }
+    })
+  }
+
+  useFrame(() => {
+    applyImpulseToBodiesInRange()
+  })
+</script>
+
+<Object3DInstance bind:object={obj} {position} />
```

Comment:
You can use this Instance to let child components bind to their parent, which would make sense when you want to bind e.g. `<TransformControls>` to it like this (try it, it's fun 😄):

```html
<Attractor position={{ x: -25, y: 10 }} range={20} strength={strengthLeft} {showHelper}>
	<TransformControls />
</Attractor>
```

So I'm proposing this component layout instead (so that the child meshes inherit the position attribute rather than get their own "copy"):

```html
<Object3DInstance
  bind:object={obj}
  {position}
>
  <slot />
  {#if showHelper}
    <Mesh
      geometry={new SphereBufferGeometry(range)}
      material={new MeshBasicMaterial({ wireframe: true })}
    />
    <Mesh
      geometry={new SphereBufferGeometry()}
      material={new MeshBasicMaterial({ color: 'black', wireframe: false })}
    />
  {/if}
</Object3DInstance>
```

Reply:
When it comes to the property `showHelper` we should consider adding a context to the component `<World>` called `threlte-rapier-world-debugging` which is a `Writable<boolean>`. The component `<Debug>` would - besides drawing the helper stuff that is included in rapier - also set this store to `true` so that you would not need to add a separate property to the `<Attractor>` component, what do you think?